packages/@mantine/form/src/resolvers/zod-resolver/zod-resolver.ts

```diff
@@ -30,7 +30,7 @@ export function zodResolver<T extends Record<string, any>>(schema: ZodSchema<T>)
 
     const results: Record<string, any> = {};
 
-    (parsed as ZodParseError).error.errors.forEach((error) => {
+    (parsed as ZodParseError).error.errors.toReversed().forEach((error) => {
       results[error.path.join('.')] = error.message;
```

Comment:
If a field has multiple errors, the first ones are overwritten by the last one. Reversing errors order makes sure that the first error message is returned in the result